### PR TITLE
Fixes inconsistencies in Neo4j integration doc

### PR DESCRIPTION
**Pull request by tomasfabian**
Fixes inconsistencies in Neo4j integration doc:
- compilation issues such as using target-typed new expressions with interface return types in queries
- MongoDB is replaced with Neo4j
- added package `HotChocolate.AspNetCore` installation instruction for the extension method `AddGraphQLServer`
- typo fix in the word annotate

### CODE REVIEW

File: website/src/docs/hotchocolate/integrations/neo4j.md [hunk 1]
```diff
@@ -15,6 +15,17 @@ To use the Neo4J integration, you need to install the package `HotChocolate.Data
 dotnet add package HotChocolate.Data.Neo4J
 ```
 
+To use the HotChocolate AspNetCore integration, you need to install the package `HotChocolate.AspNetCore`.
+
+```bash
+dotnet add Neo4j package HotChocolate.AspNetCore
```
Comment (michaelstaib) [inline website/src/docs/hotchocolate/integrations/neo4j.md:21]:
```suggestion
dotnet add package HotChocolate.AspNetCore
```

File: website/src/docs/hotchocolate/integrations/neo4j.md [hunk 2]
```diff
@@ -15,6 +15,17 @@ To use the Neo4J integration, you need to install the package `HotChocolate.Data
...
+```
+
+```csharp
+services
```
Comment (michaelstaib) [inline website/src/docs/hotchocolate/integrations/neo4j.md:25]: This seems incomplete

Comment (tomasfabian) [reply website/src/docs/hotchocolate/integrations/neo4j.md:25]: Hi @michaelstaib, the intent of this code snippet was to show why is it necessary to also add the  `HotChocolate.AspNetCore` package. Should I include something else here, too? I thought that a basic simple example will be the best in this place. But let me know if we can improve it of course.

Thank you very much for the review!

Comment (michaelstaib) [reply website/src/docs/hotchocolate/integrations/neo4j.md:25]: I think this hold section about AspNetCore is a bit confusing. Its also not necessary to have this package.
Lets kick this out and then we can merge the fixes that you have done to the docs.

**Pull request merged**